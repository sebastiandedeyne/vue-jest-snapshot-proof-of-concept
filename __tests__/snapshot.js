import { createRenderer } from 'vue-server-renderer';
import Vue from 'vue';

const renderer = createRenderer();

async function render(vm) {
    return await new Promise((resolve, reject) => {
        renderer.renderToString(vm, (error, html) => {
            if (error) {
                reject(error);
            }
            resolve(html);
        });
    });
}

it('matches a snapshot', async () => {

    const vm = new Vue({
        render(h) {
            return h('div', 'Hello world!');
        },
    });

    expect(await render(vm)).toMatchSnapshot();
});
