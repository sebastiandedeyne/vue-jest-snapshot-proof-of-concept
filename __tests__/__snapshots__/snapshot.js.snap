exports[`test matches a snapshot 1`] = `"<div server-rendered=\"true\">Hello world!</div>"`;
